service/history/handler.go

```diff
@@ -606,6 +606,36 @@ func (h *Handler) StartWorkflowExecution(ctx context.Context, request *historyse
 	return response, nil
 }
 
+func (h *Handler) ExecuteMultiOperation(
+	ctx context.Context,
+	request *historyservice.ExecuteMultiOperationRequest,
+) (_ *historyservice.ExecuteMultiOperationResponse, retError error) {
+	defer metrics.CapturePanic(h.logger, h.metricsHandler, &retError)
+	h.startWG.Wait()
+
+	namespaceID := namespace.ID(request.GetNamespaceId())
+	if namespaceID == "" {
+		return nil, h.convertError(errNamespaceNotSet)
+	}
+
+	shardContext, err := h.controller.GetShardByNamespaceWorkflow(namespaceID, request.WorkflowId)
+	if err != nil {
+		return nil, h.convertError(err)
+	}
+
+	engine, err := shardContext.GetEngine(ctx)
+	if err != nil {
+		return nil, h.convertError(err)
+	}
+
+	response, err := engine.ExecuteMultiOperation(ctx, request)
+	if err != nil {
+		return nil, h.convertError(err)
+	}
+	// TODO: shardContext.NewVectorClock()
```

Comment:
TODO?

Reply:
Yes, still a todo for follow-up work. The `StartWorkflowExecution` is doing this, and I want need to (probably) do the same thing here but haven't yet.